### DISCUSSION TITLE
Will this work? shairport-sync with RPi 4 and iPhone mobile hotspot with no Internet

### DISCUSSION THREAD

**Discussion by DINsetting** [mikebrady/shairport-sync#1353] [Q&A] [answered]
Niche Use Case:  looking to use a RPi4 with HiFiberry DAC HAT on a boat using a 12V to 5V step down converter. The goal is to improve overall sound quality (versus relying on Bluetooth) by leveraging shairport-sync AP/AP2 for audio only and be able to use the phone/mobile device to control, select audio tracks/playlists, etc.. The problem or hurdle I see is that while in most cases, I will have good cellular data connection and mobile hotspot capability (via the iPhone), I will NOT have a traditional Internet connection or LAN on the boat. 

Can what I'm trying to do be accomplished if I don't have Internet, but do have cellular/mobile hotspot capability with the iPhone? 

How do I best tackle this and where do I start? I have purchased a RPi4 model B, a HiFiBerry DAC+ Pro ADC HAT and the necessary case.

Would it be technically feasible to turn the WiFi on the Pi into an access point and connect the phone to that? Or the other way around? Been searching for a couple days, but have not found anything definitive, so here I am. 

Thank you in advance for any insight, guidance and/or advice.

**Comment by charlesomer** [comment]
I'm not sure if it would work but you might be able to connect the Raspberry Pi to the iPhone's Wi-Fi hotspot and AirPlay might appear as an option that way?

**Reply by DINsetting** [reply]
Thanks, that's what I was thinking and the path I'm on. I was just wondering if it might be possible (and more practical) to do it the other way around and have the iPhone connect to the RPi with the Pi acting as an AP, but wondering if by doing so without Internet if it would "break" cellular data on the iPhone.

**Reply by charlesomer** [reply]
It looks like there are ways to do this: https://apple.stackexchange.com/questions/173894/can-i-connect-to-wifi-but-still-use-cellular.

I think the iPhone can detect when an Wi-Fi connection has no internet and prompts to the user to "use without internet" but I'm not sure if that means mobile data still works. I guess it's worth trying and then if not, maybe try the stack overflow method?

**Comment by quantonian** [comment]
Yes, this is possible. With an access point configured on the pi you can connect directly to that network from your iphone. In order to use cellular data the network connection on the iphone needs to be configured Manually. For the network setting for the pi connection set Configure IP to Manual. Use the same settings (from Automatic) for IP and Router but set Subnet Mask to 8.8.8.8 and save. This is the only way I've been able to get this to work to use cellular data for streaming, etc.

**Comment by Siegum** [comment, marked as answer]
...just follow the instructions that mikebrady made for a car-install. It will do exactly what you are asking for...and instructions are easy to follow :-)
https://github.com/mikebrady/shairport-sync/blob/master/CAR%20INSTALL.md

**Reply by DINsetting** [reply]
Awesome, thank you!

I've spent the last week playing around with a GL-iNet travel router running with the Pi off of an Anker power bank. Works okay, but a little wonky. I have to utilize a second iDevice to act as the remote control for Volumio/Pi because you can't be connected to a WiFi network and serve a hotspot at the same time from the same device, at least not without being tethered via USB. In other words, I can't provide the Pi a hotspot and be connected to the same WiFi SSID at the same time. As soon as I connect to the WLAN, it drops the hotspot clients. Going to buy another flash card and try this method. Curious to see if I run into the same issue since the Pi will still not have a traditional Internet connection, but rather simply serving as an WLAN AP. My fear is that I will only be able to AirPlay downloaded content and not cast up high quality streams.

**Reply by Siegum** [reply]
I was able to stream apple music from my iPhone in my car with no problems, so as long as it is an iDevice and not android, I think you are good to go

**Reply by DINsetting** [reply]
Thanks again to Mike Brady and to Siegum for the nudge in the right direction. I was able to follow the 'Shairport-Sync for Car' guide and get this to work without a hitch. It's working well and will allow me to improve sound quality via WiFi versus using Bluetooth in my boat. I also really like the idea that people can simply connect to my boat's SSID and "cast" to the AirPlay target, versus futzing with trying to pair via Bluetooth. Cheers!

**Reply by noelhibbard** [reply]
I haven't read Mike's guide but I used to have an RPi in my car that ran shariport-sync and acted as a WiFi hotspot and it worked really well. The trick was to make sure the dhcp server didn't hand out a default gateway. This caused iOS to continue using LTE for internet rather than the WiFi connection. I was then able to AirPlay to my car. It worked well with multiple users in the car without messing with pairing/unpairing Bluetooth. This would especially be handy with the party scene you typically have on a boat.

**Reply by DINsetting** [reply]
> I haven't read Mike's guide but I used to have an RPi in my car that ran shariport-sync and acted as a WiFi hotspot and it worked really well. The trick was to make sure the dhcp server didn't hand out a default gateway. This caused iOS to continue using LTE for internet rather than the WiFi connection. I was then able to AirPlay to my car. It worked well with multiple users in the car without messing with pairing/unpairing Bluetooth. This would especially be handy with the party scene you typically have on a boat.

I wish there was a way to be able to **wirelessly** "tether" the RPi to the iPhone's hotspot AND create a WiFi access point while still having the ability to use LTE on the iDevice. I can't seem to get it to work. It will work if I use a travel router and connect the phone via a USB cable, but as soon as I unplug that, and try to use the Internet on the iDevice, my hotspot connection to the travel router breaks. In other words, I can't be connected to a wireless SSID that my phone's hotspot is providing the Internet to.  This would be useful with other images, like Volumio for example where you could use your phone's hotspot out on the lake to provide Internet for music streams, initiate SpotifyConnect streams and have the RPi play the stream directly from the Internet via the hotspot, etc. I can't seem to get it to work without a cable though. Sure seems like it should be possible, but something is preventing it from working correctly without a cable. Anybody know if there's something I'm missing in that scenario?

**Reply by noelhibbard** [reply]
Yeah your use case is a little different from mine. I wasn't looking to give the RPi internet access. I only wanted to take advantage of AirPlay. I also just saw Mike updated the docs to point out that AirPlay will not work if the iPhone has active hotspot connections. BUT.... you could still sort of accomplish what you are after if you had a second iDevice.

The RPi would connect to iDevice 1's hotspot which would give it internet access. The RPi would then create it's own hotspot which iDevice 2 would connect to. iDevice 1 would not be able to AirPlay but iDevice 2 could. You would need two WiFi interfaces or maybe tether iDevice 1 via USB so you don't need to get a second WiFi adapter. I know it's possible to USB tether an RPi to an iPhone because I've used one as an iPhone to ethernet bridge as a backup internet provider in a crunch before.

Oh also, if you configure the DHCP server on the RPi to not issue a default gateway then iDevice 2 would not route internet traffic through iDevice 1. Instead, iDevice 2 would use it's own LTE for internet and use WiFi just for local traffic (AirPlay).

**Reply by DINsetting** [reply]
> Yeah your use case is a little different from mine. I wasn't looking to give the RPi internet access. I only wanted to take advantage of AirPlay. I also just saw Mike updated the docs to point out that AirPlay will not work if the iPhone has active hotspot connections. BUT.... you could still sort of accomplish what you are after if you had a second iDevice.
> 
> The RPi would connect to iDevice 1's hotspot which would give it internet access. The RPi would then create it's own hotspot which iDevice 2 would connect to. iDevice 1 would not be able to AirPlay but iDevice 2 could. You would need two WiFi interfaces or maybe tether iDevice 1 via USB so you don't need to get a second WiFi adapter. I know it's possible to USB tether an RPi to an iPhone because I've used one as an iPhone to ethernet bridge as a backup internet provider in a crunch before.
> 
> Oh also, if you configure the DHCP server on the RPi to not issue a default gateway then iDevice 2 would not route internet traffic through iDevice 1. Instead, iDevice 2 would use it's own LTE for internet and use WiFi just for local traffic (AirPlay).

Yup yup, I was able to do that, but while it works, is less than ideal. What about using (2) travel routers and running one of them in 'Client Bridge Mode' with OpenWRT or DD-WRT? I have two GL-iNET Mango travel routers. Only problem is that I don't believe they support DD-WRT, but they do support OpenWRT, which I believe has the bridge mode capability. You think this might work?

See this guys blog post (the second option):
https://blogx.ch/blog/travel/travel-router-with-iphone-wireless-tethering-916/
